lib/pool.py

```diff
@@ -1,14 +1,15 @@
 import logging
 import traceback
+from typing import Optional
```

Comment:
`Optional` gets deprecated for `| None` in 3.10.  With 3.8 being already EOL and 3.9 following in 6 months we may want to upgrade our minimal python version?